### PR TITLE
Patch onfocus and onblur on HTMLElement

### LINKED ISSUE #212: onfocus property is not polyfilled
### Description
Event handlers added using properties are not polyfilled.
Example:
```
element.onfocus = (event) => { console.log(event); };
```

#### Live Demo
http://jsbin.com/solewuwuva/edit?html,output

#### Steps to Reproduce
Assign a function to `onfocus` property.

#### Expected Results
Assignment is handled by the polyfill.

#### Actual Results
Function is added as an event handler directly on the element. When it is invoked it receives a patched event and `event.currentTarget` is incorrect (`Window` in the demo).

### Browsers Affected

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Safari 9
- [ ] Safari 8
- [ ] IE 11

### Versions
- webcomponents: vX.X.X

### PR DESCRIPTION
**Pull request by TimvdLippe**
I had to store them in a special location, so I chose an obscure name which hopefully does not clash (given that we can't use `Symbol` for that.
### Reference Issue
Fixes #212

### CODE REVIEW

**Comment by TimvdLippe** [comment]
Chrome 41 is complaining here, but I have no clue why. For some reason, the propertyDescriptors are properly installed on `HTMLElement.prototype`, but once I do `Object.getOwnPropertyDescriptor(document.createElement('input'), 'onfocus')` it returns a different property. However, on the `HTMLInputElement.prototype`, the `onfocus` is my own property descriptor :thinking:

**Comment by sorvell** [comment]
In older versions of webkit (old Safari and Chrome), native element properties exist only on instances, not on their prototypes so we cannot patch prototypes. We address this for things like `firstChild` by patching instances on these browsers. We may be able to do the same thing here.

**Comment by TimvdLippe** [comment]
@sorvell Updated. It traverses through `Document.prototype` to find all the event listeners and uses ShadyData to store.

**Comment by ggriffithsIDBS** [comment]
Any estimates on time scale for this patch? Want to avoid having to fudge my own polyfill to let us get to polymer 2.0 with our legacy app

File: src/patch-builtins.js [hunk 1]
```diff
@@ -246,9 +246,34 @@ let htmlElementMixin = utils.extendAll({
...
+
+for (const property of Object.getOwnPropertyNames(Document.prototype)) {
+  const split = property.split('on');
+
+  if (split[0] === '') {
+    Object.defineProperty(htmlElementMixin, property, {
+      /** @this {HTMLElement} */
+      set: function(fn) {
+        const shadyData = ensureShadyDataForNode(this);
+        shadyData.__onCallbackListeners[property] && this.removeEventListener(split[1], shadyData.__onCallbackListeners[property]);
```
Comment (sorvell) [inline src/patch-builtins.js]: Using the `split[1]` as the event name is problematic since 'onselectionchange'.split('on') == ["", "selecti", "change"]

Comment (TimvdLippe) [reply src/patch-builtins.js]: Ah good call. Instead, I think I should array spread in the definition and then join the remainder on `on` again.

**Comment by ggriffithsIDBS** [comment]
@TimvdLippe  Question: will this patch go into the v1 and v2 releases?

**Comment by TimvdLippe** [comment]
@azakus knows. He is trying to release, but we found some breaking changes
which are blocking the release.

**Pull request merged**